nimble/host/src/ble_eatt.c

```diff
@@ -472,17 +572,25 @@ int
 ble_eatt_tx(uint16_t conn_handle, uint16_t cid, struct os_mbuf *txom)
 {
     struct ble_eatt *eatt;
+    static struct ble_l2cap_chan *channel;
     int rc;
 
     BLE_EATT_LOG_DEBUG("eatt: %s, size %d ", __func__, OS_MBUF_PKTLEN(txom));
     eatt = ble_eatt_find(conn_handle, cid);
-    if (!eatt || !eatt->chan) {
+    if (!eatt) {
+        BLE_EATT_LOG_ERROR("Eatt not available");
+        rc = BLE_HS_ENOENT;
+        goto error;
+    }
+
+    channel = ble_eatt_channel_find_by_cid(eatt, cid);
```

Comment:
This looks odd to me. If `ble_eatt_find()` takes as parameter `cid`, I would expect it will return `eatt` instance matching `cid` given. Now, when you place an array of channels into eatt so it does not simply wrap L2CAP channel, the `ble_eatt_find()` should not take `cid` as parameter IMO.